functions/src/webhooks/transcription.ts

```diff
@@ -0,0 +1,53 @@
+import * as functions from "firebase-functions"
+import { AssemblyAI } from "assemblyai"
+import { db } from "../firebase"
+import { Hearing } from "../events/types"
+import { sha256 } from "js-sha256"
+
+const assembly = new AssemblyAI({
+  apiKey: process.env.ASSEMBLY_API_KEY ? process.env.ASSEMBLY_API_KEY : ""
+})
+
+export const transcription = functions.https.onRequest(async (req, res) => {
+  if (
+    req.headers["X-Maple-Webhook"] &&
+    req.headers["webhook_auth_header_value"]
+  ) {
+    if (req.body.status === "completed") {
+      const transcript = await assembly.transcripts.get(req.body.transcript_id)
+      if (transcript && transcript.webhook_auth) {
+        const maybeEventInDb = await db
+          .collection("events")
+          .where("videoAssemblyId", "==", transcript.id)
+          .get()
+        if (maybeEventInDb.docs.length) {
+          const authenticatedEventsInDb = maybeEventInDb.docs.filter(e => {
+            const hearing = Hearing.check(e)
+            const hashedToken = sha256(
+              String(req.headers["webhook_auth_header_value"])
```

Comment:
It does occur to me that this will have security issues - we'll need to prevent users from reading `webhook_auth_header_value`. Firestore sadly doesn't let you block specific fields from returning in firestore rules - we may want to add `webhook_auth_header_value` as a field on a document in a subcollection of `transcriptions` so we can set access controls separately. This will let us make `transcriptions/${transcriptionId}` docs publically readable but the `transcriptions/${transcriptionId}/private/webhookAuth` inaccessible to client calls.

IMO this isn't a blocker for getting this PR out for testing, but is likely needed for V1.


Source: https://firebase.google.com/docs/firestore/security/rules-fields#allowing_read_access_only_for_specific_fields

Reply:
Got it. Just to make sure we're aligned, do you mean `events/${eventId}/private/webhookAuth` rather than `transcriptions/${transcriptionId}/private/webhookAuth` since I'm storing the auth token on the event?

Reply:
🤦 Yup, you're right - the auth is on the event, so it would be `events/${eventId}/private/webhookAuth`

Reply:
Cool, I'll jump on that,